src/tests/access/test_ownable_twostep.cairo

```diff
@@ -0,0 +1,346 @@
+}
+
+#[test]
+#[available_gas(2000000)]
+fn test_renounceOwnership() {
+    let mut state = setup();
+    testing::set_caller_address(OWNER());
+    state.ownable.renounceOwnership();
+
+    assert_event_ownership_transferred(OWNER(), ZERO());
+
+    assert(state.ownable.owner() == ZERO(), 'Should renounce ownership');
+}
+
+#[test]
+#[available_gas(2000000)]
+#[should_panic(expected: ('Caller is the zero address',))]
+fn test_renounceOwnership_from_zero_address() {
+    let mut state = setup();
+    state.ownable.renounceOwnership();
+}
+
+#[test]
+#[available_gas(2000000)]
+#[should_panic(expected: ('Caller is not the owner',))]
+fn test_renounceOwnership_from_nonowner() {
+    let mut state = setup();
+    testing::set_caller_address(OTHER());
+    state.ownable.renounceOwnership();
+}
+
+//
+// full two step ownership transfer (happy path)
+//
+
+#[test]
+#[available_gas(2000000)]
+fn test_full_two_step_transfer() {
+    let mut state = setup();
+    testing::set_caller_address(OWNER());
+    state.ownable.transfer_ownership(OTHER());
+
+    assert_event_ownership_transfer_started(OWNER(), OTHER());
+    assert(state.ownable.owner() == OWNER(), 'Owner should be OWNER');
+    assert(state.ownable.pending_owner() == OTHER(), 'Pending owner should be OTHER');
+
+    testing::set_caller_address(OTHER());
+    state.ownable.accept_ownership();
+
+    assert_event_ownership_transferred(OWNER(), OTHER());
+    assert(state.ownable.owner() == OTHER(), 'Owner should be OTHER');
+    assert(state.ownable.pending_owner() == ZERO(), 'Pending owner should be ZERO');
+}
+
+//
+// test accept ownership after original owner renounced
+//
+
+#[test]
+#[available_gas(2000000)]
+fn test_pending_accept_after_owner_renounce() {
+    let mut state = setup();
+    testing::set_caller_address(OWNER());
+    state.ownable.transfer_ownership(OTHER());
+
+    assert_event_ownership_transfer_started(OWNER(), OTHER());
+    assert(state.ownable.owner() == OWNER(), 'Owner should be OWNER');
+    assert(state.ownable.pending_owner() == OTHER(), 'Pending owner should be OTHER');
+
+    state.ownable.renounce_ownership();
+
+    assert_event_ownership_transferred(OWNER(), ZERO());
+    assert(state.ownable.owner() == ZERO(), 'Should renounce ownership');
+
+    testing::set_caller_address(OTHER());
+    state.ownable.accept_ownership();
+
+    assert_event_ownership_transferred(ZERO(), OTHER());
+    assert(state.ownable.owner() == OTHER(), 'Owner should be OTHER');
+    assert(state.ownable.pending_owner() == ZERO(), 'Pending owner should be ZERO');
+}
+
+//
+// Helpers
+//
+
+fn assert_event_ownership_transfer_started(
+    previous_owner: ContractAddress, new_owner: ContractAddress
+) {
+    let event = utils::pop_log::<OwnershipTransferStarted>(ZERO()).unwrap();
+    assert(event.previous_owner == previous_owner, 'Invalid `previous_owner`');
+    assert(event.new_owner == new_owner, 'Invalid `new_owner`');
+    utils::assert_no_events_left(ZERO());
```

Comment:
Since now these event members are indexed, keys should be tested similar to what is being done in ERC20 and ERC721.

Reply:
Done. Asserting it for both events `OwnershipTransferred` and `OwnershipTransferStarted`.